GitUI/CommandsDialogs/RevisionFileTreeControl.cs

```diff
@@ -287,7 +287,7 @@ protected override void OnRuntimeLoad()
 
         private IEnumerable<string> FindFileMatches(string name)
         {
-            var candidates = Module.GetFullTree(_revision.TreeGuid.ToString());
+            var candidates = _revision == null ? Enumerable.Empty<string>() : Module.GetFullTree(_revision.TreeGuid.ToString());
```

Comment:
```suggestion
            var candidates = _revision == null ? Array.Empty<string>() : Module.GetFullTree(_revision.TreeGuid.ToString());
```
`Array.Empty<T>` is a singleton, whilst `Enumerable.Empty<T>` is not.
In .NET Core it was fixed, but I don't believe it was fixed in .NET Fx.

I think the type inference should work, if not the following should work:
```suggestion
            IEnumerable<string> candidates = _revision == null ? Array.Empty<string>() : Module.GetFullTree(_revision.TreeGuid.ToString());
```

Reply:
It seems fixed in framework 4.8, see : https://referencesource.microsoft.com/#System.Core/System/Linq/Enumerable.cs,1284 or in 4.6.1: https://github.com/microsoft/referencesource/blob/4.6.2/System.Core/System/Linq/Enumerable.cs#L1161

Reply:
The link is pointing to 4.6.2, not 4.6.1